### PR TITLE
rds_snapshot_facts module

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
New rds_snapshot_facts module based on boto3

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
rds_snapshot_facts

##### ANSIBLE VERSION

```
ansible 2.6.0 (devel 4e2876be9f) last updated 2018/05/11 13:01:31 (GMT +1000)
  config file = None
  configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/will/src/opensource/ansible/lib/ansible
  executable location = /Users/will/src/opensource/ansible/bin/ansible
  python version = 2.7.14 (default, Apr 25 2018, 15:22:05) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)]

```

##### ADDITIONAL INFORMATION
Relies on aws/rds module_utils from #39990 

Based on work from #30746 and before

No tests as tests are expected to be provided by rds_snapshot module.

### CODE REVIEW

**Review by ryansb — APPROVED** [review]
LGTM, pending utils in #39990

File: lib/ansible/modules/cloud/amazon/rds_snapshot_facts.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    if instance_name:
+        params['DBInstanceIdentifier'] = instance_name
+    if snapshot_type:
+        params['SnapshotType'] = snapshot_type
+        if snapshot_type == 'public':
+            params['IsPublic'] = True
+        elif snapshot_type == 'shared':
+            params['IsShared'] = True
+
+    paginator = conn.get_paginator('describe_db_snapshots')
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/rds_snapshot_facts.py]: This doesn't support aurora snapshots. Besides that, this looks great.

Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/rds_snapshot_facts.py]: This doesn't support aurora snapshots. Besides that, this looks great.

**Review by s-hertel — CHANGES_REQUESTED** [review]

**Review by s-hertel — APPROVED** [review]

**Comment by s-hertel** [comment]
Output with an aurora snapshot:
```
ok: [localhost] => {
    "changed": false,
    "cluster_snapshots": [
        {
            "allocated_storage": 1,
            "availability_zones": [
                "ca-central-1a",
                "ca-central-1b"
            ],
            "cluster_create_time": "2018-05-17T00:13:40.223000+00:00",
            "db_cluster_identifier": "test-aurora-cluster",
            "db_cluster_snapshot_arn": "arn:aws:rds:ca-central-1:448830907658:cluster-snapshot:test-aurora-snapshot",
            "db_cluster_snapshot_identifier": "test-aurora-snapshot",
            "engine": "aurora",
            "engine_version": "5.6.10a",
            "iam_database_authentication_enabled": false,
            "kms_key_id": "arn:aws:kms:ca-central-1:448830907658:key/0950f127-beed-43e6-92d4-15e3d098030b",
            "license_model": "aurora",
            "master_username": "shertel",
            "percent_progress": 0,
            "port": 0,
            "snapshot_create_time": "2018-05-17T00:23:23.731000+00:00",
            "snapshot_type": "manual",
            "status": "creating",
            "storage_encrypted": true,
            "tags": {},
            "vpc_id": "vpc-5cd32d35"
        }
    ],
    "invocation": {
        "module_args": {
            "aws_access_key": null,
            "aws_secret_key": null,
            "db_cluster_identifier": null,
            "db_cluster_snapshot_identifier": null,
            "db_instance_identifier": null,
            "db_snapshot_identifier": null,
            "ec2_url": null,
            "profile": "shertel",
            "region": "ca-central-1",
            "security_token": null,
            "snapshot_type": null,
            "validate_certs": true
        }
    },
    "snapshots": []
}
```

**Comment by s-hertel** [comment]
bot_status

**Comment by s-hertel** [comment]
Thanks @willthames and @mikedlr! Merged to devel in 57c801c34fa16538f943fe4b8f0358f693a5039e.

**Pull request closed without merging**